python/interpret_community/explanation/explanation.py

```diff
@@ -1666,145 +1669,161 @@ def get_value(x):
-            return DatasetWrapper(value)
-        if _metadata[param] == 'DenseData':
-            return DatasetWrapper(value)
-
-    # default - no tranformation needed
-    return value
-
-
-def save_explanation(explanation):
+def save_explanation(explanation, path, exist_ok=False):
     """Serialize the explanation.
 
     :param explanation: The Explanation to be serialized.
     :type explanation: Explanation
+    :param path: The path to the directory in which the explanation will be saved. By default, must be a new directory
+        to avoid overwriting any previous explanations. Set exist_ok to True to overrule this behavior.
+    :type path: str
+    :param exist_ok: If False (default), the path provided by the user must not already exist and will be created by
+        this function. If True, a prexisting path may be passed. Any preexisting files whose names match those of the
+        files that make up the explanation will be overwritten.
+    :type exist_ok: bool
     :return: JSON-formatted explanation data.
     :rtype: str
     """
-    paramkeys = list(ExplainParams.get_serializable())
-    expldict = dict()
-    _metadata = dict()
-    for paramkey in paramkeys:
-        param = getattr(ExplainParams, paramkey)
-        if hasattr(explanation, param):
-            value = getattr(explanation, param)
+    if os.path.exists(path) and not exist_ok:
+        raise Exception('The directory specified by path already exists. '
+                        'Please pass in a new directory or set exists_ok=True.')
+    os.makedirs(path, exist_ok=True)
+
+    uploadable_properties = [
+        ExplainParams.FEATURES,
+        ExplainParams.LOCAL_IMPORTANCE_VALUES,
+        ExplainParams.EXPECTED_VALUES,
+        ExplainParams.CLASSES,
+        ExplainParams.GLOBAL_IMPORTANCE_NAMES,
+        ExplainParams.GLOBAL_IMPORTANCE_RANK,
+        ExplainParams.GLOBAL_IMPORTANCE_VALUES,
+        ExplainParams.PER_CLASS_NAMES,
+        ExplainParams.PER_CLASS_RANK,
+        ExplainParams.PER_CLASS_VALUES
+    ]
+    # TODO will need to add viz data on top of this
+    for prop in uploadable_properties:
+        if hasattr(explanation, prop) and getattr(explanation, prop) is not None:
+            value = getattr(explanation, prop)
             if isinstance(value, pd.DataFrame):
-                expldict[param] = value.values.tolist()
-                _metadata[param] = 'DataFrame'
+                value = value.values.tolist()
+                metadata = 'DataFrame'
             elif isinstance(value, DatasetWrapper):
-                expldict[param] = value.original_dataset.tolist()
-                _metadata[param] = 'DatasetWrapper'
+                value = value.original_dataset.tolist()
+                metadata = 'DatasetWrapper'
             elif isinstance(value, DenseData):
-                expldict[param] = value.original_dataset.tolist()
-                _metadata[param] = 'DenseData'
+                value = value.original_dataset.tolist()
```

Comment:
if this is iml.datatypes.DenseData, there is no original_dataset, I think you need to use .data, for example:

https://github.com/interpretml/interpret-community/blob/master/python/interpret_community/common/model_wrapper.py#L341

Reply:
but, it's a very unusual case, I don't think our explanations would have DenseData stored on them anyway...

Reply:
I was basing this on the serialization that was here before, but given that point maybe I'll just remove that section of save and load